### ISSUE TITLE
Display mail date and time instead of age in mailbox

### ISSUE THREAD

**Issue by Octofinger** [nextcloud/mail#1095]
### Feature Request

In the mailbox view, mails are sorted by age and their age is shown next to the sender name.

Currently, all mails have their age shown, which might make sense for mails that are minutes or hours old, or even up to a day. But after this, the mail age makes emails almost impossible to find. Mails that are "one month ago" can be hundreds of emails. An email that is 21 hours old requires me to do math instead of just reading the timestamp.

Suggestion:
Mark emails with:
Today's date: "Today, [hh:mm]"
Yesterday's date: "Yesterday, [hh:mm]"
Older: "[YYYY-MM-DD] [hh:mm]" (or other corresponding configurable local date format)

I will monitor this feature request, and if I can get some hints on where to look for this feature in the code, I can consider registering for contributing by making the appropriate code changes. I haven't contributed to Nextcloud Mail before but I wouldn't mind doing.

#### Summary
Display mail date and time instead of age for older emails.

**Comment by ChristophWurst** [comment]
Is this a duplicate of https://github.com/nextcloud/mail/issues/912?

**Comment by Octofinger** [comment]
No, this is about how the mail date/age is displayed in the mailbox mail list. This feature request is not related to [#912](https://github.com/nextcloud/mail/issues/912) which is a bug report that the mail date is taken from the wrong attribute.

This feature request is only related to how the mail date/age is _displayed_ in the mail list.

**Comment by ChristophWurst** [comment]
Oh, I see.

**Comment by jancborchardt** [comment]
Both relative and absolute dates have their downsides when used exclusively. This is what Geary (Gnome Mail) does and we could do that too. It works well in the space also because it is short:

- Less than 1 hour ago: "just now", "1m ago", "20m ago", etc
- Less than 12 hours ago: "1h ago", "10h ago", etc
- Same day: Timestamp like "00:33"
- Yesterday: "Yesterday"
- Less than a week ago: day of the week: "Tuesday", "Saturday", etc
- Same year: "Feb 19", "Jan 2", etc
- Older: Date like "19 Dec 2019", "2 Nov 2018" (or if that is not possible, whatever the region format like "19.12.2019")

![timestamp geary](https://user-images.githubusercontent.com/925062/75430566-1453a980-597e-11ea-843d-2e98f4342159.png)

**Comment by vipergts450** [comment]
> Is it possible to add the "time of the day" (marked in green) to the message list OR at least message details (marked in red) ?
> 
> ![b5e0fc56bcdc4b2ebb574961219f65d3 copy](https://user-images.githubusercontent.com/49558119/213559293-12107b62-9572-41e4-9efc-bd123028279d.jpg)

Yes please! Just a moment ago I was hovering over each of these places hoping that a full timestamp would appear...

**Comment by BiglifeMatt** [comment]
Are we already able to adjust the timestamp to 12 hour (am/pm)?  If not, can this be optional or driven from another setting?

**Comment by GretaD** [comment]
As mention by @PeViz here: #13005 
_"Given that NcDateTime is now widely used across Nextcloud apps, this might be a good time to revisit it."_
